### DISCUSSION TITLE
OpenLayers and flatgeobuf, problem with Events

### DISCUSSION THREAD

**Discussion by Dadej1010** [flatgeobuf/flatgeobuf#396] [Q&A] [answered]
```
import { createLoader } from 'flatgeobuf/lib/mjs/ol'; 
...
const vectorSource = new VectorSource({ strategy: bboxStrategy });

getS3Url(source.url).then(signedUrl => {
    vectorSource.setLoader(createLoader(vectorSource, signedUrl, "EPSG:4326", bboxStrategy, false))
});

vectorSource.on('featuresloadend', function (e: any) {
    console.log('layer ready', layer.getProperties().id);
});

const layer = new VectorLayer({
    source: vectorSource,
    minZoom: source.minZoom, // We only want to load the data beyond certain zoom level so that we don't load too much data
maxZoom: source.maxZoom,
    properties: {
    id: source.id     
    } 
})

layer.setStyle(getStyleFunction(source)); 
map.addLayer(layer)
```

My code above. I am trying to invoke a callback after features are loaded from VectorSource on 'featuresloadend' Event. The signedUrl points to a flatgeobuf file. flatgeobuf version 3.35, OLMaps ver: 10.2.1
event 'featuresloadstart' fires correctly and all features are visually displayed and listed by forEachFeatureAtPixel.

Does anybody have an idea what am I doing wrong?

**Comment by Dadej1010** [comment]
Why doesn't the Loader returned by createLoader emit 'featuresloadend'?

**Comment by bjornharrtell** [comment]
@Dadej1010 your issues/questions doesn't seem to be about flatgeobuf but rather OpenLayers. I will close this as unrelated to this project.

**Comment by riley-nova** [comment, marked as answer]
I was experiencing these same issues, at the moment I'm using a work around and modifying the createLoader function in my codebase. I've raised a PR which would fix this: https://github.com/flatgeobuf/flatgeobuf/pull/406

@Dadej1010 - you could just copy the createLoader function that I use in that pull request and include it in your own code, which is what I am doing currently.

**Reply by bjornharrtell** [reply]
A release including this improvement is now out, https://www.npmjs.com/package/flatgeobuf/v/3.37.0. Sorry for the delay.
